### ISSUE TITLE
[REVIEW]: swisslandstats-geopy: Python tools for the land statistics datasets from the Swiss Federal Statistical Office

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1511]
**Submitting author:** @martibosch (<a href="http://orcid.org/0000-0001-8735-9144">Martí Bosch</a>)
**Repository:** <a href="https://github.com/martibosch/swisslandstats-geopy" target ="_blank">https://github.com/martibosch/swisslandstats-geopy</a>
**Version:** 0.7.2
**Editor:** @leouieda
**Reviewer:** @weikang9009, @darribas
**Archive:** <a href="https://doi.org/10.5281/zenodo.3387312" target="_blank">10.5281/zenodo.3387312</a>

## Status

[![status](http://joss.theoj.org/papers/b6de0f096382d4dcd5d137a3f1edcb30/status.svg)](http://joss.theoj.org/papers/b6de0f096382d4dcd5d137a3f1edcb30)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/b6de0f096382d4dcd5d137a3f1edcb30"><img src="http://joss.theoj.org/papers/b6de0f096382d4dcd5d137a3f1edcb30/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/b6de0f096382d4dcd5d137a3f1edcb30/status.svg)](http://joss.theoj.org/papers/b6de0f096382d4dcd5d137a3f1edcb30)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/martibosch/swisslandstats-geopy" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@weikang9009 & @darribas, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @leouieda know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/martibosch/swisslandstats-geopy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 0.7.2
- [x] **Authorship:** Has the submitting author (@martibosch) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/martibosch/swisslandstats-geopy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 0.7.2
- [x] **Authorship:** Has the submitting author (@martibosch) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @weikang9009, @darribas it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01511/joss.01511/10.21105.joss.01511.pdf)

**Comment by leouieda** [comment]
:wave: @weikang9009 @darribas @martibosch this is the review thread for the `swisslandstats-geopy` paper. All of our communications will happen here from now on.

Both @weikang9009 and @darribas have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `openjournals/joss-reviews#1511` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within 2-4 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me here (@leouieda) or email me privately if you have any questions/concerns.

**Comment by leouieda** [comment]
:wave: Hi @weikang9009 and @darribas just checking in on the progress of the reviews. I see some issues have been opened (and resolved) in the project repository. Please let me know what your current status is and any hangups in the review that might need to be resolved.

**Comment by weikang9009** [comment]
Hi @leouieda, I am almost done with the review. @martibosch has been very responsive in resolving the issues I opened. I just opened two issues related to adding a statement of the target audience and missing DOIs in the paper.

**Comment by leouieda** [comment]
@weikang9009 thanks for the update and opening all those issues :+1: Let me know when you have a final verdict (meaning that the software passes all requirements)

**Comment by darribas** [comment]
HEllo! Sorry for the delay, getting on the review now, should be able to turn it around rather quickly!

**Comment by weikang9009** [comment]
Hi @leouieda, the package was updated and [a new version (V0.7.1) was released on github and pypi 7 days ago by @martibosch](https://github.com/martibosch/swisslandstats-geopy/releases/tag/v0.7.1). Since this most recent version does not match the one given in the review points (shown below), I am not sure how to deal with this?

> - [ ] **Version:** Does the release version given match the GitHub release (v0.6.1)?

**Comment by leouieda** [comment]
@darribas thanks for the update :+1: 

@weikang9009 that happens when there is a release while the review is ongoing. As long as the version is close and matches the Github release/tag, then it should be OK. We'll update the version information prior to publication.

**Comment by darribas** [comment]
On the authorship, @martibosch has clearly contributed to the package. In fact, he's the only contributor). The second author has not. Usually, this would point to lack of substantive contribution but I could see how this is also reasonable if the second author has provided guidance, technical support, etc. Happy to check that one off if @martibosch agrees.

**Comment by darribas** [comment]
OK that's my review for now @martibosch. Such a cool little package!!!

**Comment by martibosch** [comment]
Regarding the comment on authorship: this library has been created in the context of my PhD thesis, which is why I included my thesis supervisor as second author. Nevertheless, since this paper is not central to my thesis it is not necessary that he appears here as I am indeed the sole contributor to the library. I have therefore changed the paper accordingly in [1c8f888](https://github.com/martibosch/swisslandstats-geopy/commit/1c8f888bc15aa278de99b2bd98c928390956957a)

**Comment by leouieda** [comment]
@martibosch we don't require all authors to be package contributors since there are non-code contributions that merit authorship but aren't registered in the git log. I'll leave the final decision up to you.

**Comment by darribas** [comment]
Yes, my question wasn't a demand but much more in line with @leouieda's view. It was just raising bringing it up...

**Comment by martibosch** [comment]
well I believe that in this case, leaving it to one author is more coherent with the [JOSS autorship guidelines](https://joss.readthedocs.io/en/latest/submitting.html#authorship).

**Comment by leouieda** [comment]
:wave: @weikang9009 and @darribas please have a final look to see if all your comments have been addressed.

@darribas I see that there are some unchecked items in your reviewer checklist. Are these still in progress or have they been satisfied with the recent changes?

**Comment by darribas** [comment]
Completed now, good good to go!

**Comment by weikang9009** [comment]
Also good to go from my side!

**Comment by leouieda** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01511/joss.01511/10.21105.joss.01511.pdf)

**Comment by leouieda** [comment]
@martibosch I submitted a PR with a few minor fixes to the JOSS paper. Please let me know when you merge it and we can move on with the publication.

**Comment by martibosch** [comment]
@leouieda thank you for your corrections. I have already reviewed and merged the PR with your fixes.
Martí

**Comment by labarba** [comment]
Is this submission ready for the final checks before acceptance?

**Comment by leouieda** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01511/joss.01511/10.21105.joss.01511.pdf)

**Comment by leouieda** [comment]
@labarba thank you for the nudge, it's completely my fault that this hasn't moved along. I made one final editorial PR to the paper. After that, this should be good to go.

**Comment by leouieda** [comment]
@martibosch  I'm happy say that your paper is ready for publication in JOSS! :tada: Here are the final steps you need to take:

- Double check authors and affiliations (including ORCIDs)
- Make a release with the latest changes from the review and post the version number here.
- Archive the release on Zenodo/figshare and post the DOI here (make sure the title and author list matches the JOSS paper, if possible).

**Comment by martibosch** [comment]
> - [x] Double check authors and affiliations (including ORCIDs)

Done

> - [ ] Make a release with the latest changes from the review and post the version number here.

The latest release is 0.7.1, and all the subsequent commits have been amendments to the paper/README and including the "examples" folder in the GitHub repository (however, such folder is not included in the MANIFEST), therefore nothing would change in an hypothetical new release (well, the README would be different - but that does not seem to justify a release in semantic versioning). Shall we leave it like that, or release a patch (i.e., 0.7.2) with the above changes?

> - [ ] Archive the release on Zenodo/figshare and post the DOI here (make sure the title and author list matches the JOSS paper, if possible).

Once the second task is done, I will proceed with the third and last one.

**Comment by leouieda** [comment]
@martibosch it would be best to have a patch release if possible. You might want to take this opportunity to set the citation information for the package to the JOSS paper (the doi will be 10.21105.joss.01511) and have it registered in the release.

**Comment by martibosch** [comment]
I have released 0.7.2 and archived it to Zenodo https://doi.org/10.5281/zenodo.3387312 (with the matching title and author list).

**Comment by leouieda** [comment]
@whedon set 0.7.2 as version

**Comment by whedon** [comment]
OK. 0.7.2 is the version.

**Comment by leouieda** [comment]
@whedon set 10.5281/zenodo.3387312 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3387312" target="_blank">10.5281/zenodo.3387312</a> is the archive.

**Comment by leouieda** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1109/MCSE.2011.37 is OK
- 10.1016/j.compenvurbsys.2017.05.004 is OK
- 10.2785/132365 is OK
- 10.25080/Majora-4af1f417-011 is OK
- 10.3334/ORNLDAAC/1570 is OK
- 10.1007/s10980-019-00850-7 is OK
- 10.1101/645549 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by leouieda** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01511/joss.01511/10.21105.joss.01511.pdf)

**Comment by leouieda** [comment]
@martibosch thank you for the updates. Everything looks good to go now :confetti_ball:

**Comment by leouieda** [comment]
@openjournals/joss-eics this submission is ready for publication

**Comment by arfon** [comment]
:wave: @Kevin-Mattheus-Moerman - do you want to do the final checks on this submission?

**Comment by Kevin-Mattheus-Moerman** [comment]
@martibosch can you check/work on the following points in relation to your paper: 

- [ ] We are about to process this submission for acceptance. Can you please thoroughly read and double check the paper? Also ensure all author names and affiliations are accurate and that acknowledgements to contributors and funding sources have been added. 
- [ ] Can you remove the acronyms from your affiliations (I would keep them if they are used in the text but since they are not you can remove them). 
- [ ] In "..are rarely capable of processing such format.". Please consider using "formats" or "a format". 
- [ ] Does NumPy need its own independent reference? 
- [ ] Although usable are you able to improve the resolution of figure 1 (its a bit fuzzy)? 
- [ ] Fix the word nevertheless in"Neverhteless, although other...."
- [ ] Once you have complete the above, please run `@whedon generate pdf` here to update your proofs

**Comment by Kevin-Mattheus-Moerman** [comment]
@leouieda can you verify the above points and ping me once we are ready to proceed? Thanks

**Comment by martibosch** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01511/joss.01511/10.21105.joss.01511.pdf)

**Comment by martibosch** [comment]
* [x]  We are about to process this submission for acceptance. Can you please thoroughly read and double check the paper? Also ensure all author names and affiliations are accurate and that acknowledgements to contributors and funding sources have been added.

* [x]  Can you remove the acronyms from your affiliations (I would keep them if they are used in the text but since they are not you can remove them).

* [x]  In "..are rarely capable of processing such format.". Please consider using "formats" or "a format".
* [x]  Does NumPy need its own independent reference?

Well, I believe that since pandas, shapely and OSMnx have their own independent reference, NumPy should be treated equally. But this can be changed if you consider it more appropriate.

* [x]  Although usable are you able to improve the resolution of figure 1 (its a bit fuzzy)?

I have changed the respective code snippets in the paper so that the full extent of the canton of Vaud is plotted. I believe it might be a bit too big now. I can amend the commit with a smaller figure if you consider it more appropriate.

* [x]  Fix the word nevertheless in"Neverhteless, although other...."
* [x]  Once you have complete the above, please run `@whedon generate pdf` here to update your proofs

**Comment by leouieda** [comment]
> Well, I believe that since pandas, shapely and OSMnx have their own independent reference, NumPy should be treated equally. But this can be changed if you consider it more appropriate.

I agree that numpy deserves a citation.

> I have changed the respective code snippets in the paper so that the full extent of the canton of Vaud is plotted. I believe it might be a bit too big now. I can amend the commit with a smaller figure if you consider it more appropriate.

The figure size is fine but the resolution is a bit low (which I think was the original criticism). If you could please save it with a higher DPI, that would solve the problem.

**Comment by martibosch** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01511/joss.01511/10.21105.joss.01511.pdf)

**Comment by leouieda** [comment]
Thank you @martibosch, the figure looks much better now :+1: 

@Kevin-Mattheus-Moerman I believe all points have been addressed and we should be ready to move forward.

**Comment by Kevin-Mattheus-Moerman** [comment]
@martibosch thanks for making those changes.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1109/MCSE.2011.37 is OK
- 10.1016/j.compenvurbsys.2017.05.004 is OK
- 10.2785/132365 is OK
- 10.25080/Majora-4af1f417-011 is OK
- 10.3334/ORNLDAAC/1570 is OK
- 10.1007/s10980-019-00850-7 is OK
- 10.1101/645549 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/962

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/962, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1172603505850408965) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/963
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01511](https://doi.org/10.21105/joss.01511)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by Kevin-Mattheus-Moerman** [comment]
@arfon, still a 404 it seems, on the DOI. I'll check again later.

**Comment by arfon** [comment]
> @arfon, still a 404 it seems, on the DOI. I'll check again later.

Looks like there was a minor issue with the paper (no affiliation index) which for some reason Whedon failed silently on. I've fixed this now.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01511/status.svg)](https://doi.org/10.21105/joss.01511)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01511">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01511/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01511/status.svg
   :target: https://doi.org/10.21105/joss.01511
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01511/status.svg)](https://doi.org/10.21105/joss.01511)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-joss
